dockstore-webservice/src/main/java/io/dockstore/webservice/resources/AliasableResourceInterface.java

```diff
@@ -49,27 +52,38 @@
      */
     T getAndCheckResourceByAlias(String alias);
 
-    default T addAliases(User user, Long id, String aliases) {
-        T c = getAndCheckResource(user, id);
-        Set<String> oldAliases = c.getAliases().keySet();
-        Set<String> newAliases = Sets.newHashSet(Arrays.stream(aliases.split(",")).map(String::trim).toArray(String[]::new));
-
+    static void checkAliases(Set<String>  aliases, User user) {
         // reserve some prefixes for our own use
         String[] invalidPrefixes = {"dockstore", "doi", "drs", "trs", "dos", "wes"};
 
+        // Gather up any aliases that contain invalid prefixes
+        List<String> invalidAliases = new ArrayList<>();
         if (!user.isCurator() && !user.getIsAdmin()) {
-            for (String newAlias : newAliases) {
-                if (StringUtils.startsWithAny(newAlias, invalidPrefixes)) {
-                    throw new CustomWebApplicationException(newAlias + " starts with a reserved string, please try another alias",
-                        HttpStatus.SC_BAD_REQUEST);
-                }
-            }
+            invalidAliases = aliases.stream().filter(alias -> StringUtils.startsWithAny(alias, invalidPrefixes)).collect(Collectors.toList());
```

Comment:
Won't this disallow prefixes like `drs1`, `dosido`, etc.? Is that that the intent, or should you be checking for a colon? (I don't know).

Reply:
I am not sure; that is the way it was originally...

Reply:
@denis-yuen please comment on this, thanks!

Reply:
Yeah, that was the intent. Wanted to cast a wide net just in case. 
i.e. 
a prefix like `drs:` wouldn't block `drsv1:`, but a prefix like `drs` would block both just in case